extensions/amp-date-display/0.1/test/test-amp-date-display.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../amp-date-display';
+
+describes.realWin('amp-date-display', {
+  amp: {
+    extensions: ['amp-date-display'],
+  },
+}, env => {
+
+  let win;
+  let element;
+  let impl;
+
+  beforeEach(() => {
+    win = env.win;
+    element = win.document.createElement('amp-date-display');
+    win.document.body.appendChild(element);
+    impl = element.implementation_;
+  });
+
+  // Unfortunately, we cannot test the most interesting case of UTC datetime
+  // displayed in local, because the test would work in only one time zone.
```

Comment:
Thanks for the tip. I looked into the library and think it does not help. It can fiddle with the timer APIs. However, here we would need to change how `Date.parse` works because the result now depends on the computer's clock. I guess I could mock it but then we test nothing.

Reply:
Sounds good, thanks for looking into it.